### DISCUSSION TITLE
How to perform single tap using co-ordinates when there is no element id is available

### DISCUSSION THREAD

**Discussion by surajb0710** [appium/appium#17049] [Q&A] [answered]
Hi, 
I have only co-ordinates of the point where I want to tap on mobile and I have been trying for it, but no success. I have tried with actions class, hashmap and Jsexecutor but no success.
Can anyone pls share some code on this?

My input will be x, y coordinates only. I am using java, selenium, appium, android.

I have tried to modify below code for x, y coordinates but did not work.
((JavascriptExecutor) driver.executeScript("mobile: clickGesture", ImmutableMap.of(
    "elementId", ((RemoteWebElement) element).getId()
));

Most of the articles are filled with touchaction class which is deprecated and I could not implement w3 actions class. Also, other articles are using the element to get the coordinates which is not my scenario.
TIA.

**Comment by RSM-SBA** [comment, marked as answer]
`
public static void action_clickOnPosition(int pointA_X, int pointA_Y) {
		PointerInput finger = new PointerInput(org.openqa.selenium.interactions.PointerInput.Kind.TOUCH, "finger");
		org.openqa.selenium.interactions.Sequence clickPosition = new org.openqa.selenium.interactions.Sequence(finger, 1);
		clickPosition
		.addAction(finger.createPointerMove(Duration.ZERO, Origin.viewport(), pointA_X,pointA_Y))
		.addAction(finger.createPointerDown(MouseButton.LEFT.asArg()))
		.addAction(finger.createPointerUp(MouseButton.LEFT.asArg()));
		driver.perform(Arrays.asList(clickPosition));	}
`

**Reply by surajb0710** [reply]
You are amazing man. This worked absolutely fine. Thank you.

**Comment by sayed2** [comment]
It worked like charm. Thanks @StephanB75
